Serializer/Normalizer/ExceptionHandler.php

```diff
@@ -11,14 +11,16 @@
 
 namespace FOS\RestBundle\Serializer\Normalizer;
 
+@trigger_error(sprintf('The %s\ExceptionHandler class is deprecated since FOSRestBundle 2.8.', __NAMESPACE__), E_USER_DEPRECATED);
+
 use JMS\Serializer\Context;
 use JMS\Serializer\GraphNavigatorInterface;
 use JMS\Serializer\Handler\SubscribingHandlerInterface;
 use JMS\Serializer\JsonSerializationVisitor;
 use JMS\Serializer\XmlSerializationVisitor;
 
 /**
- * @internal since 2.8
+ * @deprecated since 2.8
```

Comment:
The error serialization in json/xml from symfony requires the symfony/serializer. If people use the jms/serializer instead, how will exceptions be serialized to json/xml?

Reply:
For people using the Symfony Serializer component, there is a new `ProblemNormalizer` that deals with `FlattenException` instances and seamlessly hooks into the `SerializerErrorRenderer`. I am currently looking into if and how we can ship a similar solution for people who prefer JMS Serializer.

Reply:
I think we can add a custom handler in this bundle, but the renderer would probably best live in the JmsSerializerBundle.

Reply:
I think it's a good thing to provide a normalizer that supports the old `{ code: ..., message: ...}` error response for BC. But maybe it's best to provide an option to opt-in to the new problem details format according to RFC 7807 that Symfony ErrorHandler uses as well, see also #2019

Reply:
We could add an option to enable/disable the `ProblemNormalizer`. If it is not registered, the `SerializerErrorRenderer` would already render the standard format.

Reply:
I have added another option. One needs to enable it to register the new `ProblemNormalizer`. Otherwise the RFC 7807 compatible normalizer provided by Symfony will do its job.

Reply:
Not sure about the naming of ProblemNormalizer. The name suggests to use the Problem Details RFC. But it's the opposite. Maybe `ThrowableNormalizer`?
Also I see correctly, people need to enable `serialize_flatten_exception` (i.e. ProblemNormalizer), to support the `fos_rest.exception.messages` config or?
Ideally, the exception message config should work either way (new style with problem details rfc or old style, with jms or symfony serializer).

Reply:
I have renamed the class to `FlattenExceptionNormalizer`. I have also changed the new config option to `flatten_exception_format` which allows to choose the format in which `FlattenException` instances are serialized (`legacy` or `rfc7807`).